src/ports/postgres/modules/stats/pred_metrics.py_in

```diff
@@ -0,0 +1,562 @@
+
+List of the various metrics output by the function:
+
+- tp is the count of correctly-classified positives
+- tn is the count of correctly-classified negatives
+- fp is the count of misclassified negatives
+- fn is the count of misclassified positives
+- tpr = tp / (tp + fn)
+- tnr = tn / (fp + tn)
+- ppv = tp / (tp + fp)
+- npv = tn / (tn + fn)
+- fpr = fp / (fp + tn)
+- fdr = 1 - ppv
+- fnr = fn / (fn + tp).
+- acc = (tp + tn) / (tp + tn + fp + fn).
+- f1 = 2* tp / (2 * tp + fp + fn).
+    """
+    elif message.lower().strip() in ['usage', 'help', '?']:
+        help_string = """
+------------------------------------------------------------
+                        USAGE
+------------------------------------------------------------
+SELECT {schema_madlib}.binary_classifier(
+    'table_in',        -- Name of the input table
+    'table_out',       -- Table name to store the metric results
+    'pred_col',        -- Column name containing prediction results
+    'obs_col',         -- Column name containing observed (actual) values
+    'grouping_cols'    -- Comma-separated list of columns to use as group-by
+);
+    """
+    else:
+        help_string = "No such option. Use {schema_madlib}.binary_classifier('usage')"
+    return help_string.format(**locals())
+
+
+def area_under_roc(table_in, table_out, pred_col, obs_col, grp_col_str=None):
+    """ Get area under ROC curve for a binary classifier.
+
+    The 'obs_col' column in 'table_in' is assumed to be a int/float column with
+    two level: 0 and 1, 1 is considered positive and 0 is negative.
+
+    The 'pred_col' is assumed to be a float column that gives the probability of
+    the prediction being positive.
+
+    """
+    grp_cols = _parse_grp_col_str(grp_col_str)
+    _validate_args(table_in, table_out, grp_cols + [pred_col, obs_col])
+    if grp_col_str:
+        grp_str = grp_col_str + ","
+        grp_by_str = "GROUP BY " + grp_col_str
+        partition_str = "PARTITION BY " + grp_col_str
+    else:
+        grp_str = grp_by_str = partition_str = ""
+
+    sql_st = """
+    CREATE TABLE {table_out} AS
+    SELECT {grp_str}
+           sum((tpr + prev_tpr) * (fpr - prev_fpr) * 0.5) AS area_under_roc
+    FROM (
+      SELECT {grp_str}
+             tpr, fpr,
+             coalesce(lag(tpr) OVER ({partition_str} ORDER BY threshold DESC),
+                      0) AS prev_tpr,
+             coalesce(lag(fpr) OVER ({partition_str} ORDER BY threshold DESC),
+                      0) AS prev_fpr
+      FROM(
+        SELECT {grp_str}
+               threshold,
+               sum(t) OVER ({partition_str} ORDER BY threshold DESC) *
+                    1.0/ NULLIF(sum(t) OVER ({partition_str}), 0) AS tpr,
+               sum(f) OVER ({partition_str} ORDER BY threshold DESC) *
+                    1.0/NULLIF(sum(f) OVER ({partition_str}),0) AS fpr
+        FROM (
+          SELECT {grp_str}
+                 {pred_col} AS threshold,
+                 sum({obs_col}) AS t,
```

Comment:
Let's add a `::int` inside the sum. That way we can also support `obs_col` as a boolean (which would be nice considering this is binary classification).